components/net/http_loader.rs

```diff
@@ -126,42 +125,62 @@ struct NetworkHttpRequestFactory {
 
 impl NetworkHttpRequestFactory {
     fn create(&self, url: ServoUrl, method: Method, headers: Headers)
-              -> Result<HyperRequest<Fresh>, NetworkError> {
+              -> Result<HyperRequest<Fresh>, LoadError> {
         let connection = HyperRequest::with_connector(method,
                                                       url.clone().into_url().unwrap(),
                                                       &*self.connector);
 
         if let Err(HttpError::Ssl(ref error)) = connection {
             let error: &(Error + Send + 'static) = &**error;
-            if let Some(&SslError::OpenSslErrors(ref errors)) = error.downcast_ref::<SslError>() {
-                if errors.iter().any(is_cert_verify_error) {
-                    let mut error_report = vec![format!("ssl error ({}):", openssl::version::version())];
-                    let mut suggestion = None;
-                    for err in errors {
-                        if is_unknown_message_digest_err(err) {
-                            suggestion = Some("<b>Servo recommends upgrading to a newer OpenSSL version.</b>");
-                        }
-                        error_report.push(format_ssl_error(err));
-                    }
-
-                    if let Some(suggestion) = suggestion {
-                        error_report.push(suggestion.to_owned());
-                    }
-
-                    let error_report = error_report.join("<br>\n");
-                    return Err(NetworkError::SslValidation(url, error_report));
-                }
-            }
+            return Err(LoadError::new(url, LoadErrorType::Ssl { reason: error.description().into() }));
         }
 
         let mut request = match connection {
             Ok(req) => req,
-            Err(e) => return Err(NetworkError::Internal(e.description().to_owned())),
+            Err(e) => return Err(
+                LoadError::new(url, LoadErrorType::Connection { reason: e.description().to_owned() })),
         };
         *request.headers_mut() = headers;
 
         Ok(request)
     }
+
+}
+
+#[derive(Debug)]
+struct LoadError {
```

Comment:
I removed this in 87979ef65eb9eb3958f1ead564a663ba2bd117ad; any reason for it to return?